shared/modules/persistence_module.go

```diff
@@ -33,6 +38,22 @@ type PersistenceModule interface {
 	HandleDebugMessage(*messaging.DebugMessage) error
 }
 
+// TreeStore is fulfilled by the treeStore to create an
+// atomic tree component for use by the peristence context.
+type TreeStore interface {
+	// Update returns the new state hash for a given height.
+	// * Height is passed through to the Update function
+	// and used by the queries against the TxIndexer and the
+	// It updates to the future but not to the past.
+	// * Passing a higher height will cause a change
+	// but repeatedly calling the same or a lower height will
+	// not incur a change.
+	Update(pgtx pgx.Tx, txi indexer.TxIndexer, height uint64) (string, error)
+	// ClearAll completely clears the state of the trees.
+	// For debugging purposes only.
```

Comment:
Let's prefact it with `Debug` (e.g. DebugClearAll) and add an `IMPROVE` to move it into a separate file with the appropriate build tag